### PR TITLE
Study unrenrollment in Glean

### PR DESCRIPTION
**Pull request by Dexterp37**
Checklist for reviewer:

- [ ] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ This means you might have to go searching through the dependency tree to get a f
 - [demographics](#demographics)
 - [enrollment](#enrollment)
 - [study-enrollment](#study-enrollment)
+- [study-unenrollment](#study-unenrollment)
```
Comment (brizental) [inline docs/metrics.md:15]: Big fan of the new name. Very clear, specially next to `study-enrollment`.

**Review by brizental — APPROVED** [review]

**Comment by rhelmer** [comment]
@Dexterp37 do you know why the integration test is failing?

File: core-addon/Core.js [hunk 2]
```diff
@@ -521,6 +533,10 @@ export default class Core {
     // Clear the list of studies user took part in.
     await this._storage.clearActivatedStudies();
 
+    // Flip upload enabled to disabled: this will trigger a
+    // deletion-request.
+    Glean.setUploadEnabled(false);
```
Comment (rhelmer) [inline core-addon/Core.js:538]: Just a general note for glean I guess, but `setUploadEnabled` generating a deletion ping feels like an unexpected side-effect :)

**Review by rhelmer — CHANGES_REQUESTED** [review]
The changes here lgtm but the test failure should be corrected.

File: core-addon/Core.js (hunk 2 above)
Comment (Dexterp37) [reply core-addon/Core.js:538]: > 
> 
> Just a general note for glean I guess, but `setUploadEnabled` generating a deletion ping feels like an unexpected side-effect :)

Heh, that's [intended](https://mozilla.github.io/glean/book/user/general-api.html#enabling-and-disabling-metrics): when disabling upload Glean respects the user intent by signaling pipeline to delete data. The Glean-owned deletion ping has a special treatment that allows it to go through even with upload disabled!

**Comment by Dexterp37** [comment]
> 
> 
> The changes here lgtm but the test failure should be corrected.

Yup, I'll take a look on Monday.

**Review by rhelmer — APPROVED** [review]

**Pull request merged**